app/src/main/java/com/example/rajadav/adidas/DataAdapter.java

```diff
@@ -0,0 +1,73 @@
+package com.example.rajadav.adidas;
+
+import android.content.Context;
+import android.support.v7.widget.RecyclerView;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.TextView;
+
+import com.example.rajadav.adidas.database.Goal;
+
+import java.util.List;
+
+
+
+public class DataAdapter extends RecyclerView.Adapter<DataAdapter.DataAdapterViewHolder> {
+
+    private List<Goal> mData;
+    private final DataAdapterOnClickHandler mClickHandler;
+
+    public interface DataAdapterOnClickHandler {
+        void onClick(Goal goal);
+    }
+
+    public DataAdapter(DataAdapterOnClickHandler clickHandler) {
+        mClickHandler = clickHandler;
+    }
+
+    public class DataAdapterViewHolder extends RecyclerView.ViewHolder implements View.OnClickListener{
```

Comment:
Lets move the declaration of this class to the end of the file, which is the general way of having inner classes